server/container/spring/src/main/java/org/apache/james/container/spring/bean/factorypostprocessor/MailboxConfigurationBeanFactoryPostProcessor.java

```diff
@@ -96,10 +112,21 @@ public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
             registry.registerAlias(mailboxIdDeserializer, "mailbox-id-deserializer");
             registry.registerAlias(mailboxIdFactory, "mailboxIdFactory");
 
+            removeMailboxManagersExceptRightSelectedOne(registry, mailbox);
+
         } catch (ConfigurationException e) {
             throw new FatalBeanException("Unable to config the mailboxmanager", e);
         }
 
     }
 
+    private void removeMailboxManagersExceptRightSelectedOne(BeanDefinitionRegistry registry, String selectedMailboxManager) {
```

Comment:
In fact, we have some kind of mailboxManager based on the kind of databases that we chose. We defines those mailboxManager in the xml file and it's loaded before the *PostProcessor. That's why I have to remove all bean definitions of other mailboxManager except the one we choose.